urllib3/fields.py

```diff
@@ -31,45 +31,93 @@ def format_header_param(name, value):
         The name of the parameter, a string expected to be ASCII only.
     :param value:
         The value of the parameter, provided as a unicode string.
+    :ret:
+        An RFC-2231-formatted unicode string.
     """
     if not any(ch in value for ch in '"\\\r\n'):
-        result = '%s="%s"' % (name, value)
+        result = u'%s="%s"' % (name, value)
         try:
             result.encode('ascii')
         except (UnicodeEncodeError, UnicodeDecodeError):
             pass
         else:
             return result
-    if not six.PY3 and isinstance(value, six.text_type):  # Python 2:
+
+    if not six.PY3:  # Python 2:
```

Comment:
It was unnecessary, as the method doc requires that `value` be a unicode string. If we did allow non-unicode-data through here, we would need the encoding also, so it can be passed to `encode_rfc2231`. Right now, the formatted result is hardcoded to display `utf-8` as the encoding.